fission-core/library/Fission/Web/Auth/Token/UCAN/Resource/Types.hs

```diff
@@ -50,5 +56,53 @@ instance FromJSON Resource where
       Just parsed -> return parsed
       Nothing     -> fail "Does not match any known Fission resource"
 
-instance Display (NotFound Resource) where
-  display _ = "No UCAN resource provided (closed UCAN)"
+instance DelegationSemantics Resource where
+  canDelegate rsc rscProof =
+    case (rsc, rscProof) of
+      (FissionFileSystem path, FissionFileSystem proofPath) ->
+        case (isPrivate path, isPrivate proofPath, pathSubset path proofPath) of
+          (True, True, _) ->
+            bitwiseContains proofPath path
+
+          (_, _, True) ->
+            True
+
+          _ ->
+            False
+
+      (RegisteredDomain _, RegisteredDomain Complete) ->
+        True
+
+      (FissionApp _, FissionApp Complete) ->
+        True
+
+      _ ->
+        rsc == rscProof
```

Comment:
This is code adapted from a previous function `Fission.Web.Auth.Token.JWT.Proof.WNFS.compareSubsets`